js/hub.md

```diff
@@ -16,30 +14,20 @@ import { Hub } from '@aws-amplify/core';
 
 ## Working with the API
 
-### dispatch()
-
-You can dispatch an event with `dispatch` function:
-```javascript
-Hub.dispatch('auth', { event: 'signIn', data: user }, 'Auth');
-```
+### Listening for messages
 
-### listen()
+`Hub.listen(channel: string | RegExp, callback)` is used to listen for messages that have been dispatched. You must provide either a named `channel` or a regular expression, along with a callback. In the case of a regular expression only dispatches which contain a `message` in their payload will be matched against your pattern. You can add multiple listeners to your application for different channels or patterns to listen for, or trap generic events and perform your own filtering.
```

Comment:
So if I understand correctly here we allow to filter by `message` with a regexp, but if they do that they cannot filter by channel. I think these two methods of filtering events could be confused as they are passed into `listen` via the same argument, but actually match different event fields.

Listen for channel:
```js
Hub.listen('mychannel') // --> matches the "channel" field
Hub.listen(/mymessage/) // --> matches the "message" field
```

I would expect the regexp to match channel name instead

Reply:
Not exactly, you can still filter by channel just by calling listener separately. It's just in a single API call you can do one or the other. We reviewed the API with customers and our DAs and received good feedback that it was clear.